util/db/cluster.go

```diff
@@ -87,8 +87,11 @@ func (db *db) ListClusters(ctx context.Context) (*appv1.ClusterList, error) {
 	}
 	hasInClusterCredentials := false
 	for i, clusterSecret := range clusterSecrets {
-		cluster := *secretToCluster(clusterSecret)
-		clusterList.Items[i] = cluster
+		cluster, err := secretToCluster(clusterSecret)
+		if err != nil {
+			log.Errorf("could not unmarshal cluster secret: %v", err)
```

Comment:
I'm not quite sure (here as well as below) whether we should pass the value of `err` to the Logger, since it _may_ contain some confidential information?

Is the error content required for users to troubelshoot, or shall we just omit the details?